docs/dependencies.rst

```diff
@@ -0,0 +1,189 @@
+=====================================
+How Test Dependencies Work In ReFrame
+=====================================
+
+Before going into details on how ReFrame treats test dependencies, it is important to understand how tests are actually treated and executed by the runtime.
+Normally, a ReFrame test will be tried for different programming environments and different partitions within the same ReFrame run.
+These are defined in the test's :func:`__init__` method, but it is not this original object that is being executed by the regression test pipeline.
+The following figure explains in more detail the process:
+
+.. figure:: _static/img/reframe-test-cases.svg
+  :align: center
+  :alt: How ReFrame loads and schedules tests for execution.
+
+When ReFrame loads a test from the disk it unconditionally constructs it executing its :func:`__init__` method.
+The practical implication of this is that your test will be instantiated even if it will not be run on the current system.
+After all the tests are loaded, they are filtered based on the current system and any other criteria (such as programming environment, test attributes etc.) specified by the user (see `Filtering of Regression Tests <running.html#filtering-of-regression-tests>`__ for more details).
+After the tests are filtered, ReFrame creates the actual `test cases` to be run. A test case is essentially a tuple consisting of the test, the system partition and the programming environment to try.
+The test that goes into a test case is essentially a `clone` of the original test that was instantiated upon loading.
+This ensures that the test case's state is not shared and may not be reused in any case.
+Finally, the generated test cases are passed to a `runner` that is responsible for scheduling them for execution based on the selected execution policy.
+
+Dependencies in ReFrame are defined at the test level using the :func:`depends_on` function, but are projected to the test cases space.
+We will see the rules of that projection in a while.
+The dependency graph construction and the subsequent dependency analysis happens also at the level of the test cases.
```

Comment:
```suggestion
The dependency graph construction and the subsequent dependency analysis happen also at the level of the test cases.
```

Reply:
This is plural, because I have two subjects: "The dependency graph construction and the subsequent dependency analysis."